app/src/main/java/br/org/cesar/knot_setup_app/view/scan/ScanFragment.java

```diff
@@ -33,19 +39,62 @@
 public class ScanFragment extends Fragment implements ViewModel {
 
     private ListView deviceListView;
+    private TextView feedbackMessage;
 
     private ScanAdapter adapter;
     private Presenter presenter;
     private List<BluetoothDevice> deviceList;
     private UUID service;
     private Handler handler;
 
+    private Context context;
+    private BroadcastReceiver bluetoothBroadcastReceiver;
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        context = getContext().getApplicationContext();
+    }
+
+
+
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.fragment_list, container,false);
 
         deviceList = new ArrayList<>();
         adapter = new ScanAdapter(getActivity(), R.layout.item_device, deviceList);
         deviceListView = view.findViewById(R.id.list);
+        feedbackMessage = view.findViewById(R.id.searching_for_gateway_text);
+
+        //broadcast receiver that monitors the bluetooth state
+
+        bluetoothBroadcastReceiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            final String action = intent.getAction();
+            if (action.equals(BluetoothAdapter.ACTION_STATE_CHANGED)) {
+                final int state = intent.getIntExtra(BluetoothAdapter.EXTRA_STATE, BluetoothAdapter.ERROR);
+                switch(state) {
+                    case BluetoothAdapter.STATE_OFF:
```

Comment:
Will do, this is a debug log to monitor the system internal state. Since the log is needed I will move it insed the presenter. Is there any reason why it shouldn't be in the view?

Reply:
There is nothing wrong with it being in the view, the point is about adding unneeded logic (switch/case) just because of debugging/logging.

Notice that you already have the same switch/case in the presenter and the ONLY purpose of the one in the view is to "ease" debugging. Using the code suggested above you'd still know the state (only need to go in the class and check the corresponding value) and no need for the switch/case in the view. Also, if it's necessary you can use the one in the presenter to print specific log messages regarding the exact state.